### PR TITLE
Implement backward for fixed_batch_normalization

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

File: tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py [hunk 1]
```diff
@@ -30,9 +30,22 @@ def setUp(self):
         self.x = numpy.random.uniform(-1, 1, (7, 3)).astype(numpy.float32)
         self.gy = numpy.random.uniform(-1, 1, (7, 3)).astype(numpy.float32)
 
+        self.test = False
+
+    def set_avg(self):
+        mean = self.x.mean(axis=self.aggr_axes)
+        var = self.x.var(axis=self.aggr_axes)
+        if cuda.get_array_module(self.link.avg_mean) != numpy:
```
Comment (delta2323) [inline tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py]: How about use `self.xp`?

**Comment by delta2323** [comment]
LGTM except the comment. But I would like @beam2d to check the implementation because he left this feature as TODO. Does this simple solution works as you imagined?

**Comment by beam2d** [comment]
Can you implement backprop for mean and var? I think it is confusing to silently cut the backward computation without any errors.

**Comment by okuta** [comment]
I fixed. Please review this.

File: tests/chainer_tests/functions_tests/normalization_tests/test_batch_normalization.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+
+def _batch_normalization(expander, gamma, beta, x, mean, var):
+    mean = mean[expander]
+    std = numpy.sqrt(var)[expander]
+    y_expect = (gamma[expander] * (x - mean) / std + beta[expander])
+    return y_expect
+
+
+
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/normalization_tests/test_batch_normalization.py]: Remove one redundant empty line.

File: tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py [hunk 3]
```diff
@@ -11,19 +12,53 @@
...
 
-        mean = self.x.mean(axis=self.aggr_axes, keepdims=True)
-        std = numpy.sqrt(
-            self.x.var(axis=self.aggr_axes, keepdims=True) + self.link.eps)
+        mean = self.mean[self.expander]
+        if self.test:
+            std = numpy.sqrt(self.var[self.expander])
+        else:
+            std = numpy.sqrt(self.var[self.expander] + self.link.eps)
         y_expect = self.gamma * (self.x - mean) / std + self.beta
```
Comment (delta2323) [inline tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py]: Could you make the calculation of `y_expect` a function?

**Comment by delta2323** [comment]
LGTM :+1:

**Pull request merged**